raven/init.lua

```diff
@@ -0,0 +1,372 @@
+      conf.trace_level = 2
+   else
+      conf.trace_level = conf.trace_level + 1
+   end
+
+   trace_level = conf.trace_level
+   exception[1].stacktrace = backtrace(trace_level)
+
+   local payload = {
+      exception = exception,
+      message = exception[1].value,
+      culprit = self:get_culprit(trace_level),
+   }
+
+   -- because whether tail call will or will not appear in the stack back trace
+   -- is different between PUC-lua or LuaJIT, so just avoid tail call
+   local id, err = self:send_report(payload, conf)
+   return id, err
+end
+
+--- Send a message to Sentry.
+-- See [reference](https://docs.sentry.io/clientdev/interfaces/message/).
+--
+-- @function Raven:captureMessage
+-- @param message the message, usually a raw string
+-- @param conf    capture configuration table, see @{report_conf}
+-- @return        On success, return event id. If not success, return nil and
+--  an error string.
+-- @usage
+-- local rvn = ravennew(...)
+-- local id, err = rvn:captureMessage("simple message",
+--     { tags = { foo = "bar", abc = "def" }})
+function raven_mt:captureMessage(message, conf)
+   if not conf then
+      conf = { trace_level = 2 }
+   elseif not conf.trace_level then
+      conf.trace_level = 2
+   else
+      conf.trace_level = conf.trace_level + 1
+   end
+
+   local payload = {
+      message = message,
+      culprit = self:get_culprit(conf.trace_level),
+   }
+
+   local id, err = self:send_report(payload, conf)
+   return id, err
+end
+
+--- Alias of @{Raven:captureException}.
+-- @function Raven:capture_exception
+raven_mt.capture_exception = raven_mt.captureException
+
+--- Ailas of @{Raven:captureMessage}.
+-- @function Raven:capture_message
+raven_mt.capture_message = raven_mt.captureMessage
+
+local function merge_tables(msg, root)
+   if not root then
+      return msg
+   elseif not msg then
+      return root
+   end
+
+   -- both table exist, merge root into msg
+   for k, v in pairs(root) do
+      msg[k] = msg[k] or v
+   end
+   return msg
+end
+
+--- Send directly a report to Sentry.
+-- This is an internal function, you should not call it directly, use
+-- @{Raven:captureException} or @{Raven:captureMessage} instead.
+--
+-- @function Raven:send_report
+-- @param json table to be sent. Don't need to fill `event_id`, `timestamp`,
+--  `tags` and `level`.
+-- @param conf capture configuration table, see @{report_conf}
+-- @return        On success, return event id. If not success, return nil and
+--  an error string.
+function raven_mt:send_report(json, conf)
+   local event_id = uuid4()
+
+   -- TODO: Why is this line commented out?
+   --json.project   = self.project_id,
+
+   if not json then
+      json = self.json
+      if not json then
+         return
+      end
+   end
+
+   json.event_id  = event_id
+   json.timestamp = iso8601()
+   json.level     = self.level
+   json.platform  = "lua"
+   json.logger    = self.logger
+
+   if conf then
+      json.tags = merge_tables(conf.tags, self.tags)
```

Comment:
It merge `self.tags` (the instance tags) into `conf.tags` (tags for that particular message) so there is not much risk here unless the caller reuses the tables across calls. I'll update the docs to make clear that reusing tables isn't supported.